src/main/java/de/tum/in/www1/artemis/repository/metis/ConversationRepository.java

```diff
@@ -0,0 +1,39 @@
+package de.tum.in.www1.artemis.repository.metis;
+
+import java.util.List;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import de.tum.in.www1.artemis.domain.metis.Conversation;
+
+/**
+ * Spring Data repository for the Conversation entity.
+ */
+@SuppressWarnings("unused")
+@Repository
+public interface ConversationRepository extends JpaRepository<Conversation, Long> {
+
+    @Query("""
+             SELECT DISTINCT conversation FROM Conversation conversation
+             LEFT JOIN FETCH conversation.conversationParticipants conversationParticipant
+             WHERE conversation.id = :#{#conversationId}
+             AND conversationParticipant.conversation.id = conversation.id
+            """)
+    Conversation findConversationById(@Param("conversationId") Long conversationId);
+
+    @Query("""
```

Comment:
According to the server guidelines, we should avoid using nested queries. Maybe the following reference can help you resolve this nested query.
Reference: https://docs.artemis.ase.in.tum.de/dev/guidelines/server/#avoid-the-usage-of-sub-queries

Reply:
Here, I am aware of the reference in the docs and have already utilized it in previous developments. But for this specific case if I follow the suggestion in the link, I only fetch the conversationParticipant object of that particular user. What I need is if there is a conversation with the current user's conversationParticipant object, fetch that conversation with all conversationParticipants of that object, whether they belong to that user or not. I hope I could explain clearly :grimacing:.

Reply:
@Hialus would you have a suggestion for this?

Reply:
Could you do `AND conversationParticipant.user.id = :#{#userId}` here?

Reply:
I initially tried it before committing the code. The issue here is that if I do it that way, the query does not fetch other conversationParticipants of that conversation. They are also needed. (i.g. for displaying the name of the user we are conversating with)

Reply:
I have found an issue in my initial query where I did not need a sub-query and fixed it. Thanks for reminding on not to use sub-queries.